lib/Target/SubstraitPB/Export.cpp

```diff
@@ -226,8 +226,21 @@ FailureOr<std::unique_ptr<Plan>> exportOperation(PlanOp op) {
     if (failed(rel))
       return failure();
 
+    // Handle `Rel`/`RelRoot` cases depending on whether `names` is set.
     PlanRel *planRel = plan->add_relations();
-    planRel->set_allocated_rel(rel.value().release());
+    if (std::optional<Attribute> names = relOp.getFieldNames()) {
+      auto root = std::make_unique<RelRoot>();
+      root->set_allocated_input(rel->release());
+
+      auto namesArray = names->cast<ArrayAttr>().getAsRange<StringAttr>();
```

Comment:
Free standing cast form? (they marked the others as deprecated so it should be emitting warnings now)

Reply:
Thanks! I had grepped for `.cast` so I missed `->cast` 😉